src/stores/NearbyLocationsStore.js

```diff
@@ -1,58 +1,53 @@
 // @flow
 
-import type { ObservableMap } from 'mobx';
-import type { EntityID } from 'brewskey.js-api';
-import type { NearbyLocation } from '../types';
+import type { Coordinates, NearbyLocation } from '../types';
+import type { IComponentStore } from './types';
 
-import { action, computed, observable, runInAction } from 'mobx';
+import { action, computed, observable } from 'mobx';
+import { LoadObject } from 'brewskey.js-api';
 import CommonApi from '../CommonApi';
 
-class NearbyLocationsStore {
-  @observable _nearbyLocationsById: ObservableMap<NearbyLocation> = new Map();
+class NearbyLocationsStore implements IComponentStore {
+  @observable
+  nearbyLocationLoader: LoadObject<Array<NearbyLocation>> = LoadObject.empty();
+  @observable _coordinates: ?Coordinates = null;
+
+  initialize = () => {
+    CommonApi.subscribe(this._recomputeNearbyLocations);
+  };
+
+  dispose = () => {
+    CommonApi.unsubscribe(this._recomputeNearbyLocations);
+  };
 
   @action
-  fetchAll = async (): Promise<Array<NearbyLocation>> => {
-    try {
-      const coordinates = await new Promise(
-        (resolve, reject: (error: PositionError) => void) =>
-          // eslint-disable-next-line no-undef
-          navigator.geolocation.getCurrentPosition(
-            (position: Position): void =>
-              resolve({
-                latitude: position.coords.latitude,
-                longitude: position.coords.longitude,
-              }),
-            (error: PositionError) => reject(error),
-          ),
-      );
-
-      const nearbyLocations = await CommonApi.fetchNearbyLocations(coordinates);
-
-      runInAction(() => {
-        this._nearbyLocationsById.merge([
-          ...nearbyLocations.map((nearbyLocation: NearbyLocation): [
-            EntityID,
-            NearbyLocation,
-          ] => [nearbyLocation.id, nearbyLocation]),
-        ]);
-      });
-
-      return nearbyLocations;
-    } catch (error) {
-      // todo add error handling
-      return [];
-    }
+  setCoordinatesAndFetch = (coordinates: Coordinates) => {
+    this._coordinates = coordinates;
+    this._recomputeNearbyLocations();
   };
 
   @computed
   get all(): Array<NearbyLocation> {
```

Comment:
If you return a LoadObject it could be cleaner
```
return this._coordinatesLoader.map(coordinates =>
  CommonApi.fetchNearbyLocations(coordinates),
);
```

Reply:
hm..but i still can't avoid _recomputeNeabylocationLoader() on subsribe()
this means that if I return LoadObject here I'll don't need this getter at all. I just can use store.nearbyLcationsLoader directly.